### PR TITLE
SAAS-1158: mui-profile-page

### PR DESCRIPTION
**Pull request by MaksimMalakhovSimbirsoft**

### CODE REVIEW

File: src/pages/Profile/Profile.styles.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ export const getStyles = (theme: GrafanaTheme) => {
     `,
     platformAccessTokenDescription: css`
       color: ${colors.text};
-      margin: ${spacing.sm} 0 ${spacing.xl} ${spacing.xxs};
+      margin: ${spacing.sm} 0 ${spacing.xl} 0;
```
Comment (fabio-silva) [inline src/pages/Profile/Profile.styles.ts:67]: We're not supposed to use Grafana's styling anymore, check Tomas' PR for an idea on spacing from MUI.

File: src/pages/Profile/Profile.tsx [hunk 2]
```diff
@@ -1,27 +1,47 @@
...
 import { getPlatformAccessToken, copyToClipboard } from 'core';
 import { PrivateLayout } from 'components/Layouts';
 import { getStyles } from './Profile.styles';
 import { Messages } from './Profile.messages';
 import { CONNECT_PMM_DOC_LINK } from './Profile.constants';
+import { UserData, UserDataError } from './Profile.types';
 
-const { required, maxLength } = validators;
-const nameValidators = [required, maxLength(MAX_NAME_LENGTH)];
+const LoaderButton = styled(Button)(() => ({
```
Comment (fabio-silva) [inline src/pages/Profile/Profile.tsx:17]: I believe we should instead use Button's `sx` prop to style it.

File: src/pages/Profile/Profile.tsx [hunk 3]
```diff
@@ -33,94 +53,165 @@ export const ProfilePage: FC = () => {
...
   );
 
-  const handleCopyToClipboard = useCallback(async () => {
-    await copyToClipboard(platformAccessToken);
-    toast.success(Messages.copySuccessful);
-  }, [platformAccessToken]);
+  const handleInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+
+    if (value.length > MAX_NAME_LENGTH) {
```
Comment (fabio-silva) [inline src/pages/Profile/Profile.tsx:68]: Let's not do this. We're not supposed to remove react-final-form and do our own validation. RFF and Yup already do that for us.

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: src/pages/Profile/Profile.tsx [hunk 4]
```diff
@@ -33,94 +53,165 @@ export const ProfilePage: FC = () => {
...
+              disabled={pending}
+              margin="normal"
+              fullWidth
+              size="small"
+              value={userData.firstName || ''}
+              onChange={handleInput}
+              error={userData.firstName === '' || !!error.firstNameError}
+              helperText={error.firstNameError}
+              InputLabelProps={{
+                // @ts-ignore
```
Comment (nicolalamacchia) [inline src/pages/Profile/Profile.tsx:120]: don't use @ts-ignore

**Review by nicolalamacchia — CHANGES_REQUESTED** [review]

File: src/pages/Profile/Profile.tsx (hunk 2 above)
Comment (MaksimMalakhovSimbirsoft) [reply src/pages/Profile/Profile.tsx]: @fabio-silva Done

File: src/pages/Profile/Profile.styles.ts (hunk 1 above)
Comment (MaksimMalakhovSimbirsoft) [reply src/pages/Profile/Profile.styles.ts]: @fabio-silva Done, I ve added m,p props from MUI, and used it not in pixels

File: src/pages/Profile/Profile.tsx (hunk 3 above)
Comment (MaksimMalakhovSimbirsoft) [reply src/pages/Profile/Profile.tsx]: @fabio-silva Done, I ve returned React final form and used input props for handling textfields

File: src/pages/Profile/Profile.tsx [hunk 5]
```diff
@@ -33,94 +53,165 @@ export const ProfilePage: FC = () => {
...
+    } else {
+      setActivityChanged(true);
+    }
+
+    setError((prevState) => ({ ...prevState, ...{ [name]: !value } }));
+    setUserData((prevState) => ({ ...prevState, ...{ [name]: value } }));
+  };
+
+  const handleUpdateProfileSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
```
[the remark this answers is not in the document]
Comment (MaksimMalakhovSimbirsoft) [reply src/pages/Profile/Profile.tsx]: @nicolalamacchia Done, I ve removed it and used props from react final form

File: src/pages/Profile/Profile.tsx (hunk 4 above)
Comment (MaksimMalakhovSimbirsoft) [reply src/pages/Profile/Profile.tsx]: @nicolalamacchia How can I avoid ts ignoring ? I ve heard about making wrapper around component, I wrapped my textfield from MUI by field from react final form , but it is not working( we should discuss how can I solve it

**Review by TomasBlazek3PG — CHANGES_REQUESTED** [review]

File: src/pages/Profile/Profile.tsx (hunk 4 above)
Comment (nicolalamacchia) [reply src/pages/Profile/Profile.tsx]: pushed a commit that remove them, PTAL

File: src/pages/Profile/Profile.styles.ts [hunk 6]
```diff
@@ -1,72 +1,66 @@
...
-    `,
-  };
+export const styles = {
+  wrapper: {
+    display: 'flex',
+    justifyContent: 'center',
+    width: '100%',
+  },
+  form: {
+    flex: '1',
```
[the remark this answers is not in the document]
Comment (MaksimMalakhovSimbirsoft) [reply src/pages/Profile/Profile.styles.ts:10]: @TomasBlazek3PG Done, changed to number

File: package.json [hunk 7]
```diff
@@ -33,8 +33,11 @@
     "playwright:run-headed": "npx playwright test --headed"
   },
   "dependencies": {
+    "@emotion/react": "^11.10.4",
+    "@emotion/styled": "^11.10.4",
```
[the remark this answers is not in the document]
Comment (MaksimMalakhovSimbirsoft) [reply package.json:37]: @TomasBlazek3PG Done

**Pull request closed without merging**